pkg/kubelet/dockertools/manager.go

```diff
@@ -1223,10 +1224,61 @@ func (dm *DockerManager) runContainerInPod(pod *api.Pod, container *api.Containe
 		// containers to avoid daemons or POD containers are killed by oom killer.
 		util.ApplyOomScoreAdj(containerInfo.State.Pid, userContainerOomScoreAdj)
 	}
+	addNDotsOption(containerInfo.ResolvConfPath, container.Name)
 
 	return kubeletTypes.DockerID(id), err
 }
 
+func addNDotsOption(resolvFilePath, containerName string) error {
+	if len(resolvFilePath) == 0 {
+		glog.Errorf("DNS ResolvConfPath is empty for container %s", containerName)
+		return nil
+	}
+
+	if !fileExists(resolvFilePath) {
+		glog.Errorf("DNS ResolvConfPath does not exist and could not be updated. Path:%s, Container:%s. ", resolvFilePath, containerName)
+		return nil
+	}
+
+	glog.Infof("DNS ResolvConfPath exists:%s for container:%s. Will attempt to add ndots:3 option.", resolvFilePath, containerName)
+
+	ndotsOption := "options ndots:3\n"
+	for i := 0; i < 10; i++ {
+		err := appendToFile(resolvFilePath, ndotsOption)
+		if err == nil {
+			glog.Infof("resolv.conf updated with ndots option for container:%s", containerName)
+			return nil
+		}
+		time.Sleep(time.Second)
+	}
+	glog.Errorf("resolv.conf could not be updated for container:%s", containerName)
+	return nil
+}
+
+func fileExists(fileName string) bool {
+	_, err := os.Stat(fileName)
+	return os.IsExist(err)
+}
+
+func appendToFile(filePath, stringToAppend string) error {
+	f, err := os.OpenFile(filePath, os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		glog.Errorf("Error opening file %s for appending. err: %v", filePath, err)
+		return err
+	}
+	defer f.Close()
+
+	//err = syscall.Flock(f.Fd(), LOCK_EX)
+
+	_, err = f.WriteString(stringToAppend)
+	if err != nil {
+		glog.Errorf("Error writing to file %s. err: %v", filePath, err)
```

Comment:
I don't think we should log this except on our last loop - the caller should log it